metrics/go_metrics.go

```diff
@@ -781,3 +871,142 @@ func (me *Metrics) RecordAdsCertReq(success bool) {
 func (me *Metrics) RecordAdsCertSignTime(adsCertSignTime time.Duration) {
 	me.adsCertSignTimer.Update(adsCertSignTime)
 }
+
+func (me *Metrics) RecordModuleDuration(labels ModuleLabels, duration time.Duration) {
+	mm, err := me.getModuleMetric(labels)
+	if err != nil {
+		return
+	}
+
+	// Module metrics
+	mm.DurationTimer.Update(duration)
+	// Account-Module metrics
+	if labels.PubID != "" {
+		if aam, ok := me.getAccountMetrics(labels.PubID).moduleMetrics[labels.Module]; ok {
+			aam.DurationTimer.Update(duration)
+		}
+	}
```

Comment:
I'm wondering if this should always be executed or if it should only be executed in a debug mode. If it's only supposed to be executed in a debug mode, perhaps it should be a separate account-specific metrics function. I'll talk to the team.

Reply:
Yes, I've answered to this question above. Some debug switcher for account module metrics will be added